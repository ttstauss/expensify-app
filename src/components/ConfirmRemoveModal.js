import React from 'react'
import Modal from 'react-modal'

const ConfirmRemoveModal = props => (
  <Modal
    isOpen={props.modalIsOpen}
    contentLabel="Remove?"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Are you sure?</h3>
    <p className="modal__subtitle">You will not be able to recover this expense.</p>
    <div className="button-group">
      <button className="button button--secondary" onClick={props.handleCancelRemove}>Cancel</button>
      <button className="button button--warning" onClick={props.handleConfirmRemove}>Yes, delete it!</button>
    </div>
  </Modal>
)

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app')

export default ConfirmRemoveModal