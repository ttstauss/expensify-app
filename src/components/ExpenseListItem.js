import React from 'react'
import { Link } from 'react-router-dom'

export default ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>
  </div>
)