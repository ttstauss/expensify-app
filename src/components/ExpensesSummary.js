import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenses }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
        {hiddenExpenses > 0 && <p className="page-header__message">Not showing {hiddenExpenses} expenses</p>}
      </div>
    </div>
  )
}

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  const hiddenExpenses = state.expenses.length - visibleExpenses.length
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
    hiddenExpenses: hiddenExpenses
  }
}

export default connect(mapStateToProps)(ExpensesSummary)