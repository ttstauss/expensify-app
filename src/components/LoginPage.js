import React from 'react'
import { connect } from 'react-redux'
import { startGoogleLogin, startFacebookLogin } from '../actions/auth'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGoogle, faFacebookF } from '@fortawesome/free-brands-svg-icons'

export const LoginPage = ({ startGoogleLogin, startFacebookLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">Expensify</h1>
      <p>It's time to get your expenses under control</p>
      <button className="button button--google" onClick={startGoogleLogin}>
        <FontAwesomeIcon className="social-icon" icon={faGoogle} />
        Login with Google
      </button>
      <button className="button button--facebook button--last" onClick={startFacebookLogin}>
        <FontAwesomeIcon className="social-icon" icon={faFacebookF} />
        Login with Facebook
      </button>
    </div>
  </div>
)

const mapDispatchToProps = dispatch => ({
  startGoogleLogin: () => dispatch(startGoogleLogin()),
  startFacebookLogin: () => dispatch(startFacebookLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)