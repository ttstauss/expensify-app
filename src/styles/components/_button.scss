.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

.button--warning {
  background: #c62828;
}

.button-group {
  display: flex;
  justify-content: space-between;
}

%button--social-shared {
  margin: 0 0 $s-size 0;
  transition: background .3s ease;
  width: 100%;
}

.button--google {
  @extend %button--social-shared;
  background: #db3236;
  &:hover {
    background: lighten(#db3236, 7%)
  }
}

.button--facebook {
  @extend %button--social-shared;
  background: #3B5998;
  &:hover {
    background: lighten(#3B5998, 7%)
  }
}

.button--last {
  margin: 0;
}