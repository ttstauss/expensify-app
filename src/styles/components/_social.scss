.social-icon {
  margin: 0 $s-size 0 0;
}