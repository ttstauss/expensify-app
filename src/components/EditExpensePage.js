import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import ConfirmRemoveModal from '../components/ConfirmRemoveModal'

export class EditExpensePage extends React.Component {
  state = {
    modalIsOpen: false,
    confirmRemove: false
  }
  onSubmit = expense => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  openModal = () => {
    this.setState(() => ({ modalIsOpen: true }))
  }
  handleConfirmRemove = () => {
    this.setState(() => ({
      modalIsOpen: false,
      confirmRemove: true
    }))
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/')
  }
  handleCancelRemove = () => {
    this.setState(() => ({ modalIsOpen: false }))
  }
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <div className="content-body">
            <ExpenseForm
              expense={this.props.expense}
              onSubmit={this.onSubmit}
            />
            <div>
              <button className="button button--secondary" onClick={this.openModal}>Remove Expense</button>
            </div>
          </div>
        </div>
        <ConfirmRemoveModal
          modalIsOpen={this.state.modalIsOpen}
          handleCancelRemove={this.handleCancelRemove}
          handleConfirmRemove={this.handleConfirmRemove}
        />
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)