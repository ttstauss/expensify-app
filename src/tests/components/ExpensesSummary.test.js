import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('should render ExpensesSummary correclty with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={235} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correclty with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={23512340987} />)
  expect(wrapper).toMatchSnapshot()
})