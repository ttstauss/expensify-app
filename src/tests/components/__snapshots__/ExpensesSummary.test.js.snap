// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correclty with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correclty with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
